### PR TITLE
Feature branch: Integrate IPC Solidity contracts

### PR DESCRIPTION
**Pull request by cryptoAtwill**
# Update
This branch is becoming the base feature branch for the integration of the IPC Solidity actors with the IPC agent.
 
# Config update PR
Update the config to a new format that supports both FVM and EVM. Sample format:
```toml
[server]
json_rpc_address = "127.0.0.1:3030"

[[subnets]]
id = "/r123"
network_name = "test"

[subnets.config]
network_type = "fvm"
gateway_addr = "f01"
jsonrpc_api_http = "http://127.0.0.1:3030/rpc/v1"
accounts = ["f01", "f01"]

[[subnets]]
id = "/r1234"
network_name = "test2"

[subnets.config]
network_type = "evm"
provider_http = "http://127.0.0.1:3030/rpc/v1"
registry_addr = "0x6be1ccf648c74800380d0520d797a170c808b624"
gateway_addr = "0x6be1ccf648c74800380d0520d797a170c808b624"
private_key = "0x6BE1Ccf648c74800380d0520D797a170c808b624"
accounts = ["0x6be1ccf648c74800380d0520d797a170c808b624", "0x6be1ccf648c74800380d0520d797a170c808b624"]
```
All direct field access are changed to getter functions.

### CODE REVIEW

File: src/config/deserialize.rs [hunk 1]
```diff
@@ -90,3 +116,23 @@ where
         .collect();
     addrs.map_err(D::Error::custom)
 }
+
+/// A serde deserialization method to deserialize a list of eth account strings into a vector of
+/// [`Address`].
```
Comment (adlrocha) [inline src/config/deserialize.rs:121]: This `Address` is a Filecoin Address? Do you mind elaborating a bit so it is more understandable?

**Review by adlrocha — APPROVED** [review]
Amazing job! LGTM. I added a few suggestions but with those settled feel free to merge.

File: Cargo.toml [hunk 2]
```diff
@@ -83,9 +85,11 @@ fil_actors_runtime = { git = "https://github.com/consensus-shipyard/fvm-utils",
 ipc-sdk = { git = "https://github.com/consensus-shipyard/ipc-actors.git", branch = "adlrocha/subnetid-refactor" }
 ipc-subnet-actor = { git = "https://github.com/consensus-shipyard/ipc-actors.git", features = [], branch = "adlrocha/subnetid-refactor" }
 ipc-gateway = { git = "https://github.com/consensus-shipyard/ipc-actors.git", features = [], branch = "adlrocha/subnetid-refactor" }
-ipc_ipld_resolver = { git = "https://github.com/consensus-shipyard/ipc-ipld-resolver.git", branch = "main" }
+#ipc_ipld_resolver = { git = "https://github.com/consensus-shipyard/ipc-ipld-resolver.git", branch = "main" }
```
[the remark this answers is not in the document]
Comment (cryptoAtwill) [reply Cargo.toml:88]: removed

File: src/config/mod.rs [hunk 3]
```diff
@@ -32,25 +32,30 @@ pub const JSON_RPC_VERSION: &str = "2.0";
...
-id = "/r123/t01"
-gateway_addr = "f064"
-network_name = "child"
-jsonrpc_api_http = "http://127.0.0.1:1250/rpc/v1"
-auth_token = "YOUR TOKEN"
-accounts = ["t01"]
+    [server]
+    json_rpc_address = "127.0.0.1:3030"
+
+    [[subnets]]
```
[the remark this answers is not in the document]
Comment (cryptoAtwill) [reply src/config/mod.rs:38]: this is from the naming of the Config. We are calling the field `subnets`, we have to change that to `subnet` for it to work. But that would appear quite strange when reading the code.

**Pull request merged**